src/main/java/com/post/parcel_registration/controller/ParcelDeliveryController.java

```diff
@@ -0,0 +1,25 @@
+package com.post.parcel_registration.controller;
+
+import com.post.parcel_registration.model.Parcel;
+import com.post.parcel_registration.services.ParcelDeliveryService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+public class ParcelDeliveryController {
+
+    private ParcelDeliveryService parcelDeliveryService;
+
+    @Autowired
+    public ParcelDeliveryController(ParcelDeliveryService parcelDeliveryService) {
+        this.parcelDeliveryService = parcelDeliveryService;
+    }
+
+    @RequestMapping(value = "/registerParcel", method = RequestMethod.POST)
+    public ResponseEntity<?> registerParcel(@RequestBody Parcel parcel) {
```

Comment:
Parcel is not DTO!! You must not cast Entity as dto. It's absolutely two different layers in application. But you use one class for two purposes.